### PR TITLE
doc: add 0 arguments Function type constructor definition #262

### LINKED ISSUE #262: functions without parameters
Hi,

I've been reading [Type signature notation]() and the definition of `->` function type constructor.

But reading about laws specification I found something like:

- `empty :: Monoid m => () -> m`

However, something like `() -> a` does not satisfy the definition of type constructed by function type constructor:

Notice that say:
> When used in this way, the syntax is: (<input-types>) -> <output-type>, where <input-types> comprises **two or more comma–space** (,)-separated type representations.

If I'm not mistaken, the definition of function type constructor should be redefined.

### PR DESCRIPTION
**Pull request by xgbuils**
Closes #262

Sorry, maybe it was not necessary duplicate the PR.

### CODE REVIEW

**Comment by joneshf** [comment]
It's not a big deal since this is a relatively small change. But in larger changes it gets kind of hard to keep context across PRs.

For other people that might be confused this is followup from #263

File: README.md [hunk 1]
```diff
@@ -75,13 +75,13 @@ id="sanctuary-types-return">[1](#sanctuary-types)</sup>
...
+    left argument is the input type and the right argument is the output type.
     - `->`'s input type can be a grouping of types to create the type of a
-      function which accepts more than one argument (i.e. has arity greater
-      than one). When used in this way, the syntax is: `(<input-types>) ->
-      <output-type>`, where `<input-types>` comprises two or more comma–space
-      (`, `)-separated type representations.
+    function which accepts zero or more arguments. The syntax is:
+    `(<input-types>) -> <output-type>`, where `<input-types>` comprises zero or
+    more comma–space (`, `)-separated type representations and parens may be
+    ommited for unary functions.
```
Comment (davidchambers) [inline README.md]: Please keep the original indentation:

```markdown
    - Lorem ipsum dolor sit amet, consectetur adipiscing elit. Duis
      nunc diam, sagittis sed nisi ultrices, placerat condimentum dolor.
      Mauris fermentum lectus faucibus nunc malesuada dictum. Nunc quis
      massa lacinia, commodo ex non, dictum enim.
```

Comment (davidchambers) [inline README.md]: `s/ommited/omited/`

Comment (rjmk) [inline README.md]: omitted?

Comment (davidchambers) [inline README.md]: Quite right, Rafe! :laughing:

File: README.md [hunk 2]
```diff
@@ -75,13 +75,13 @@ id="sanctuary-types-return">[1](#sanctuary-types)</sup>
...
-      gives the type of functions between these two type arguments.
+    - `->` is an _infix_ type constructor that takes two type arguments where
+      left argument is the input type and the right argument is the output type.
     - `->`'s input type can be a grouping of types to create the type of a
-      function which accepts more than one argument (i.e. has arity greater
-      than one). When used in this way, the syntax is: `(<input-types>) ->
-      <output-type>`, where `<input-types>` comprises two or more comma–space
-      (`, `)-separated type representations.
+      function which accepts zero or more arguments. The syntax is:
+      `(<input-types>) -> <output-type>`, where `<input-types>`comprises zero
```
Comment (davidchambers) [inline README.md]: Nit: We're missing a space before _comprises_.

File: README.md [hunk 3]
```diff
@@ -75,13 +75,13 @@ id="sanctuary-types-return">[1](#sanctuary-types)</sup>
...
+      or more comma–space (`, `)-separated type representations and parens may
```
Comment (davidchambers) [inline README.md]: Nit: The space after the comma will need to be a NO-BREAK SPACE in order to prevent it from being collapsed when rendered as HTML.

Comment (xgbuils) [inline README.md]: I don't know how to put no-break space for markdown. Does the same apply to these expressions?

> 1. `t(u.traverse(F, x => x))` is equivalent to `u.traverse(G, t)` for any

> 3. `u.traverse(Compose, x => new Compose(x))` is equivalent to

> `new Compose(u.traverse(F, x => x).map(x => x.traverse(G, x => x)))

> `M.chainRec((next, done, v) => p(v) ? d(v).map(done) : n(v).map(next), i)`

> 3. `A.of(y).ap(u)` is equivalent to `u.ap(A.of(f => f(y)))` (interchange)

Comment (joneshf) [inline README.md]: You shouldn't have to fix this. You didn't change this part of the line.

If it needs to change, let's do it in a separate PR.

Comment (davidchambers) [inline README.md]: You're quite right, Hardy.

Xavier, these changes look good. Would you mind squashing your commits?

Comment (xgbuils) [inline README.md]: For sure. Done!
